fastembed/sparse/minicoil.py

```diff
@@ -219,18 +219,14 @@ def query_embed(
         """
         Encode a list of queries into list of embeddings.
         """
-        yield from self._embed_documents(
-            model_name=self.model_name,
-            cache_dir=str(self.cache_dir),
-            documents=query,
-            providers=self.providers,
-            cuda=self.cuda,
-            device_ids=self.device_ids,
-            k=self.k,
-            b=self.b,
-            avg_len=self.avg_len,
-            is_query=True,
-        )
+        if isinstance(query, str):
+            query = [query]
+
+        if not hasattr(self, "model") or self.model is None:
+            self.load_onnx_model()
```

Comment:
I don't like that we need to make it explicit here

Reply:
I don't see a big problem in it, I can revert though